### PR TITLE
Mrc 1607 Run report page: user can set changelog message and type

### PR DESCRIPTION
**Pull request by carbowizzy**
This PR branched out of the local branch https://github.com/vimc/orderly-web/pull/292 which I worked on. While I agree it has pending selenium test, I am determined to resolve the issue much later and hence why the branch is still open.

This PR adds changeLog message and type to run report page. Please note that the label changed as per the comments in the mockup.

### CODE REVIEW

File: src/app/static/src/tests/components/runReport/runReport.test.ts [hunk 1]
```diff
@@ -568,4 +568,72 @@ describe("runReport", () => {
...
+    });
+
+    it("it does not render changelog message and type if a report is not selected", () => {
+        const changelogTypes =  ["internal", "public"]
+        const wrapper = mount(RunReport, {
+            propsData: {
+                metadata: {
+                    git_supported: false,
+                    changelog_types: changelogTypes,
+                    selectedReport: "report"
```
Comment (mwoodbri) [inline src/app/static/src/tests/components/runReport/runReport.test.ts:603]: This line is confusing and should be removed - `metadata` will never include `selectedReport`, and this test is about the case where a report _isn't_ seletced.

File: src/customConfigTests/src/test/kotlin/org/vaccineimpact/orderlyweb/customConfigTests/RunReportPageTests.kt [hunk 2]
```diff
@@ -92,28 +92,27 @@ class RunReportPageTests : SeleniumTest()
         assertThat(tab.findElement(By.tagName("p")).text).isEqualTo("Report logs coming soon!")
     }
 
-
     //TODO: This test case should be revisited as soon as test data is updated.
-    @Test
+
     @Ignore
+    @Test
     fun `can fill parameter textField`()
```
Comment (mwoodbri) [inline src/customConfigTests/src/test/kotlin/org/vaccineimpact/orderlyweb/customConfigTests/RunReportPageTests.kt:99]: This change seems to remove an existing test, but I guess the intention is to add a new one? If so then please ensure that the original is retained.

File: src/app/static/src/js/components/runReport/runReport.vue [hunk 3]
```diff
@@ -89,7 +109,9 @@
                 runningStatus: "",
                 runningKey: "",
                 disableRun: false,
-                parameterValues: []
+                parameterValues: [],
+                changeLogMessageValue: "",
+                changeLogTypeValue: ""
```
Comment (mwoodbri) [inline src/app/static/src/js/components/runReport/runReport.vue:114]: Like other values, I think these fields should be reset when the selected report changes

File: src/app/static/src/js/components/runReport/runReport.vue [hunk 4]
```diff
@@ -39,7 +39,27 @@
...
                 <label for="params-component" class="col-sm-2 col-form-label text-right">Parameters</label>
-                <parameter-list id="params-component" @getParams="getParameterValues" :params="parameterValues"></parameter-list>
+                <parameter-list id="params-component" @getParams="getParameterValues"
+                                :params="parameterValues"></parameter-list>
+            </div>
+            <div v-if="showChangelog">
+                <div id="changelog-message" class="form-group row">
+                    <label for="changelogMessage" class="col-sm-2 col-form-label text-right">Changelog Message</label>
+                    <div class="col-sm-6">
+                        <textarea class="form-control" id="changelogMessage" v-model="changeLogMessageValue"
```
Comment (mwoodbri) [inline src/app/static/src/js/components/runReport/runReport.vue:49]: It's currently possible to leave the changelog message blank even if you've selected a changelog type. I suggest that the message should be required if you select a type.

**Review by mwoodbri — CHANGES_REQUESTED** [review]
Looks good for the most part - just a couple of corrections/improvements to be made

File: src/customConfigTests/src/test/kotlin/org/vaccineimpact/orderlyweb/customConfigTests/RunReportPageTests.kt [hunk 5]
```diff
@@ -92,28 +92,27 @@ class RunReportPageTests : SeleniumTest()
         assertThat(tab.findElement(By.tagName("p")).text).isEqualTo("Report logs coming soon!")
     }
 
-
     //TODO: This test case should be revisited as soon as test data is updated.
-    @Test
+
     @Ignore
+    @Test
```
Comment (EmmaLRussell) [inline src/customConfigTests/src/test/kotlin/org/vaccineimpact/orderlyweb/customConfigTests/RunReportPageTests.kt:98]: This test has changed but is still being ignored, should it be?

File: src/app/static/src/js/components/runReport/runReport.vue (hunk 4 above)
Comment (EmmaLRussell) [reply src/app/static/src/js/components/runReport/runReport.vue:49]: ...conversely it shouldn't be possible to add a message without specifying a type, as that will fail when it gets to orderly server. I wouldn't mind if the first changelog message type was auto-selected so there was always some type. But might be better as Mark says to require both to be populated if one is (it is ok to not include a changelog message at all).

We'd have to validate that before submitting, or enabling Run button. In that case, we'd probably want some message to the user to indicate missing information.

Comment (mwoodbri) [reply src/app/static/src/js/components/runReport/runReport.vue:49]: Sorry - yes - this is a better solution given that new changelog types can't be added (in v1 at least). I guess we just need to ensure that empty changelog messages aren't sent in mrc-2161.

**Review by mwoodbri — APPROVED** [review]
Looks good to go

**Review by EmmaLRussell — APPROVED** [review]

**Pull request merged**